paimon-common/src/main/java/org/apache/paimon/CoreOptions.java

```diff
@@ -1076,6 +1076,14 @@ public class CoreOptions implements Serializable {
                     .withDescription(
                             "Initial buckets for a partition in assigner operator for dynamic bucket mode.");
 
+    public static final ConfigOption<Integer> DYNAMIC_BUCKET_MAX_BUCKETS_PER_ASSIGNER =
+            key("dynamic-bucket.max-buckets-per-assigner")
```

Comment:
I think we should limit total buckets?

Reply:
This is also my initial purpose, but I found that the bucket generation logic in `BucketAssigner` is a instance behavior and could be parallelized, every assigner handles their own buckets. I have no idea about total buckets limits and maybe we could introudce it later, WDYT?

Reply:
I found that `dynaimc-bucket.initial-bucket` is also `assigner` granularity behavior, to make a consistence, maybe we should also use `dynamic-bucket.max-buckets` :)

Reply:
I still think introducing a max bucket is better.
The option should be very easy to understand.

Reply:
+1
Add option should be easy to user understand.

Reply:
reasonable enough, I have found a way to implement total max bucket feature, thanks for reminding that :)